exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -237,7 +237,7 @@ def _translate_to_prometheus(
                 label_keys = []
                 label_values = []
 
-                for key, value in number_data_point.attributes.items():
+                for key, value in sorted(number_data_point.attributes.items()):
```

Comment:
Forgive me for my lack of understanding of how Prometheus backend works but it looks like they accept a list of keys and values and they expect the indices of the pairs (key, value) to line up to represent a label. I'm not sure how sorting the original dictionary solves this issue? Isn't the logic simply pairing off each key/value?

Reply:
Indeed, but how the current logic works:
- Get label_keys and labels_values, for every iteration over metrics we also store labels_values in labels_valuess
- When iterating is done we label_keys, list containing order of keys of only last iteration
- labels_valuess with all label values in a nested list
- When creating a prometheus metric we specify labels at creation time only once, then prometheus expects all label values to have the correct order (when not using kwargs)

So to make sure label_keys is always consistent (not including missing keys), my solution was to always sort this.

Reply:
It also fixes the problem of generating duplicates metrics, because label keys have a different order. This happens because this code is used to filter duplicates later on:
```
pre_metric_family_ids.append(
    "|".join(
        [
            metric_name,
            metric_description,
            "%".join(label_keys),
            metric.unit,
        ]
    )
)
```